### DISCUSSION TITLE
TSS assignment and p2g analysis

### DISCUSSION THREAD

**Discussion by rojinsafavi** [GreenleafLab/ArchR#1313] [Questions / Documentation] [answered]
Hi @rcorces 

Does assigning an alternative TSS for a gene influence the p2g correlation analysis?

For a specific gene(NRG1) ArchR assigns the longest transcript as the main one for NRG1, and uses that transcripts' TSS. but when I look at NRG1, I see that another isoform is more expressed (the one circled)

Even though the assigned TSS is different from what I visually see, the inferred expression of NRG1 is in the correct cluster (when I do the integration).

I was wondering if the TSS assignment could potentially influence the peak2gene analysis in this case?

<img width="1181" alt="Screen Shot 2022-03-01 at 11 04 49 PM" src="https://user-images.githubusercontent.com/19545198/156236570-19f29e95-7a1c-4392-b2e9-b7eaecb02670.png">

**Comment by rojinsafavi** [comment]
@rcorces  I would very much appreciate it if I could get you advice into this matter, thank you so much again

**Comment by rojinsafavi** [comment]
I think since addReproduciblePeakSet uses promoter region (2000,100), if a TSS isn't assigned to the isoform of interest (the one with the highest accessibility at its TSS), this may influence the peak calling, and as a result that might influence the p2g analysis?

**Comment by rcorces** [comment, marked as answer]
I'm sorry for not replying sooner. The genes of interest come from the geneAnnotation object. To keep analysis straightforward, only one start and stop is used for each gene. Not every TSS is used. So the positions being used for addPeakToGeneLinks are the positions of the genes shown by `getGenes(ArchRProj)`. If those dont fit your analysis, you would have to edit the gene annotation information and re-run the analysis (starting from GeneScoreMatrix creation).

**Comment by rojinsafavi** [comment]
Thanks @rcorces this might be a bug, I tried fixing NRG1 manually, I changed  NRG1 start and end based on the isoform that I was interested in, and when I looked at the new grange it looks fine, I gave the new grange to addGeneScoreMatrix, and checked the genes again using getGenes,  and I noticed that it hasn't been fixed(NRG1 start and end are not what I defined). Please let me know if this is a bug, I can post it in the issue section if it is, thanks again!

```
getgenes = getGenes(projHeme2)
df = as.data.frame(getgenes)
df[df$symbol %in% c('NRG1'),]$start = 32548635
df[df$symbol %in% c('NRG1'),]$end = 32764407
getgenes0 = makeGRangesFromDataFrame(df,keep.extra.columns=T)
df = as.data.frame(getgenes0)
df[df$symbol %in% c('NRG1'),]
```
<img width="502" alt="Screen Shot 2022-03-12 at 7 49 31 PM" src="https://user-images.githubusercontent.com/19545198/158025929-1cd1c59e-2dfd-42ca-a78c-7215a4cfea65.png">

```projHeme2 = addGeneScoreMatrix(input = projHeme2,genes = getgenes0, force = T)```
```
getgenes = getGenes(projHeme2)
df = as.data.frame(getgenes)
df[df$symbol %in% c('NRG1'),]
```

<img width="505" alt="Screen Shot 2022-03-12 at 7 53 32 PM" src="https://user-images.githubusercontent.com/19545198/158026054-8a9ad2b4-2884-4721-aa6b-5c0d6405760f.png">

**Reply by rojinsafavi** [reply]
I worked around the above issue by giving the new grange to the ArchR obj as follow:

projHeme2@geneAnnotation$genes = getgenes0

Thanks again

**Reply by rcorces** [reply]
That is expected.

`getGenes()` looks at `ArchRProj@geneAnnotation$genes` which you actually havent changed.
When you run `addPeak2GeneLinks()` it uses the features (genes) that are returned by the following:
```
  ArrowFiles <- getArrowFiles(ArchRProj)
  geneSet <- .getFeatureDF(ArrowFiles, useMatrix, threads = threads)
  geneStart <- GRanges(geneSet$seqnames, IRanges(geneSet$start, width = 1), name = geneSet$name, idx = geneSet$idx)
```
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/IntegrativeAnalysis.R#L1061
where `geneStart` is the position used for p2g analysis.

In this case, it is taking the features from `GeneIntegrationMatrix` which are in turn derived from the features in `GeneScoreMatrix`
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/RNAIntegration.R#L248

So by changing the genes used in `GeneScoreMatrix` like you've done, you should change the features used for p2g analysis.

What I'm not sure about is if there will be any issues because your gene features dont match the genes in your `geneAnnotation` but I think it should be ok. The other solution would be to change the BSGenome object used for the `geneAnnotation`.
